packages/data-point/lib/reducer-types/resolve.js

```diff
@@ -45,26 +46,66 @@ function getResolveFunction (reducer) {
  * @param {Object} manager
  * @param {Accumulator} accumulator
  * @param {Reducer} reducer
+ * @param {Array|String|Number} key - id for reducer stack traces if an error is thrown
  * @returns {Promise<Accumulator>}
  */
-function resolveReducer (manager, accumulator, reducer) {
+function resolveReducer (manager, accumulator, reducer, key) {
   // this conditional is here because BaseEntity#resolve
   // does not check that lifecycle methods are defined
   // before trying to resolve them
   if (!reducer) {
     return Promise.resolve(accumulator)
   }
 
-  const resolve = getResolveFunction(reducer)
-  // NOTE: recursive call
-  const result = resolve(manager, resolveReducer, accumulator, reducer)
+  // storing this in case we need it for the catch block, since we
+  // can't trust it won't be overwritten in the accumulator object
+  const value = accumulator.value
+  const result = Promise.try(() => getResolveFunction(reducer))
+    // NOTE: recursive call
+    .then(resolve => resolve(manager, resolveReducer, accumulator, reducer))
```

Comment:
wrapping `getResolveFunction` so that `resolveReducer` will always return a promise, and errors thrown from this function will always have the `_stack` property